BLUEBUTTON-169: Currency expressed as valueMoney should be valueQuantity

jcgeer opened the pull request:
A bug was reported of our EOB resources returning extensions for Primary
Payer Paid Amount field as a valueMoney when the FHIR spec states the
value of an extension cannot be a valueMoney. This changes it to a
valueQuantity.

jcgeer committed:
BLUEBUTTON-169: Currency expressed as valueMoney should be valueQuantity

A bug was reported of our EOB resources returning extensions for Primary
Payer Paid Amount field as a valueMoney when the FHIR spec states the
value of an extension cannot be a valueMoney. This changes it to a
valueQuantity.

jcgeer committed:
BLUEBUTTON-369: Removing the JSON encoding; not part of task.

karlmdavis commented:
Before looking at the code, let's discuss whether we even _want_ to merge this change, since it'll be backwards-incompatible (and thanks for pulling the before-and-after JSON to verify that, @jcgeer !).

And another issue to look at is: this can't be the only place where we're making that mistake, can it? Don't we have tons of `valueMoney` extensions? If we do decide to fix it, we should fix it everywhere.

Did @cgreim and/or @ekivemark have opinions? Personally, I'd suggest that we can't deploy this change without first getting approval from all of our production devs. (Or deploying it as part of a later "v2" version of our API.)

jcgeer committed:
BLUEBUTTON-359: Updates

1. Multiple other fields from the EOBs were also using valueMoney in an
extension - replaced those with valueQuantity.
2. With the replacement, some TransformerUtils functions were no longer
in use - removed those.
3. Not sure how I fat-fingered the JIRA ticket number so many times.
Should definitely be 359 (not 369 or 169 as the PR and previous commits
say).

karlmdavis commented on bluebutton-server-app/src/main/java/gov/hhs/cms/bluebutton/server/app/stu3/providers/CarrierClaimTransformer.java:
```diff
@@ -61,7 +61,7 @@ private static ExplanationOfBenefit transformClaim(CarrierClaim claimGroup) {
 				Optional.of(claimGroup.getClaimTypeCode()));
 
-		TransformerUtils.addAdjudicationTotal(eob, CcwCodebookVariable.PRPAYAMT,
+		TransformerUtils.addValueQuantity(eob, CcwCodebookVariable.PRPAYAMT,
```
I'm not a fan of how this effectively renames the helps-you-understand-things method name of "`addAdjudicationTotal`" to the generic and not-terribly-helpful "`addValueQuantity`". Similar issue with the test assertion methods.

I know that's just a side effect of how this refactoring happened, but I'd recommend fixing it.

jcgeer replied:
Can do. Would these various amounts still be considered adjudications (in which case I'll put back the old name)? I don't entirely understand how "adjudication" relates to these various field amounts, and haven't seen anything that explains that.

karlmdavis replied:
Yes. "Adjudication" is a dumb word for, "decision on coverage/payment that the insurer (i.e. CMS) made." So it covers deductibles, copays, payments, etc.

jcgeer replied:
Ah, that makes a bit more sense then in that context. I've gone ahead and already made that update anticipating that would be the case. Though it looks like we'll be putting off merging this PR anytime soon.

ekivemark commented:
I am ok with deferring this to a V2 API. We will want to do R4 of FHIR early next year anyway and that would be the natural place to make the change,.

- Mark


> On Nov 9, 2018, at 11:34 AM, Karl M. Davis <notifications@github.com> wrote:
> 
> Before looking at the code, let's discuss whether we even want to merge this change, since it'll be backwards-incompatible (and thanks for pulling the before-and-after JSON to verify that, @jcgeer <https://github.com/jcgeer> !).
> 
> And another issue to look at is: this can't be the only place where we're making that mistake, can it? Don't we have tons of valueMoney extensions? If we do decide to fix it, we should fix it everywhere.
> 
> Did @cgreim <https://github.com/cgreim> and/or @ekivemark <https://github.com/ekivemark> have opinions? Personally, I'd suggest that we can't deploy this change without first getting approval from all of our production devs. (Or deploying it as part of a later "v2" version of our API.)
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/CMSgov/bluebutton-data-server/pull/121#issuecomment-437416177>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AAGXcA9VhupNGcbvypzpNWDtsUM4ZWCDks5uta6GgaJpZM4YWu8J>.
>

jcgeer committed:
BLUEBUTTON-359: Renaming method

addAdjudicationTotal is a more helpful name than addValueQuantity

jcgeer committed:
BLUEBUTTON-359: Fixing comments

Old comment info I had missed, updated to reflect the method name.

karlmdavis commented:
Closing this PR, as the project has now been migrated into a new monorepo: https://github.com/CMSgov/beneficiary-fhir-data
